packages/api-queries/src/site-users.ts

```diff
@@ -24,3 +29,9 @@ export const siteUsersQuery = ( siteId: number ) =>
 		queryKey: [ 'site', siteId, 'users', 'list' ],
 		queryFn: () => fetchSiteUsers( siteId ),
 	} );
+
+export const siteUsersWpcomQuery = ( siteId: number, role?: string ) =>
+	queryOptions( {
+		queryKey: [ 'site', siteId, 'wpcom-users', 'wpcom', role ],
```

Comment:
Nit: I don't see us needing this extra layer in the query key, but it's not a huge deal

```suggestion
		queryKey: [ 'site', siteId, 'wpcom-users', role ],
```

Having `'wpcom'` there implies maybe there'd be other categories of `wpcom-users` in the future?